ts/sbd-server/src/msg.test.ts

```diff
@@ -0,0 +1,128 @@
+import { describe, expect, assert, it, beforeAll, afterAll } from 'vitest';
+import {
+  Msg,
+  MsgLbrt,
+  MsgLidl,
+  MsgAreq,
+  MsgAres,
+  MsgSrdy,
+  MsgKeep,
+  MsgNone,
+  MsgForward,
+} from './msg.ts';
+import { ed } from './ed.ts';
+
+describe('Msg', () => {
+  it('lbrt', async () => {
+    const orig = new MsgLbrt(42);
+    expect(orig.type()).equals('lbrt');
+    expect(orig.limit()).equals(42);
+    const parsed = Msg.parse(orig.encoded());
+    expect(parsed.type()).equals('lbrt');
+    if (parsed instanceof MsgLbrt) {
+      expect(parsed.limit()).equals(42);
+    } else {
+      throw 'invalid type';
+    }
```

Comment:
You could instead use
```suggestion
    assert(parsed instanceof MsgLbrt);
    expect(parsed.limit()).equals(42);
```

or if the TS compiler does not like that
```
import assert from "node:assert";
    assert(parsed instanceof MsgLbrt);
    expect(parsed.limit()).equals(42);
```

Reply:
I indeed tried just asserting first, and indeed the ts compiler didn't accept that as enough to type the further calls. I don't really want to bring in asserts from multiple different libraries...